lib/pos_blockchain.go

```diff
@@ -291,13 +301,51 @@ func (bc *Blockchain) updateCurrentView(desoBlock *MsgDeSoBlock) {
 
 func (bc *Blockchain) GetUncommittedTipView() (*UtxoView, error) {
 	// Connect the uncommitted blocks to the tip so that we can validate subsequent blocks
-	panic("GetUncommittedTipView: Not implemented yet")
+	highestCommittedBlock, committedBlockIndex := bc.getHighestCommittedBlock()
+	if highestCommittedBlock == nil {
+		// TODO: do we just want to connect all the blocks in best chain then?
+		// This is an edge case we'll never hit in practice since all the PoW blocks
+		// are committed.
+		return nil, errors.New("GetUncommittedTipView: No committed blocks found")
+	}
+	utxoView, err := NewUtxoView(bc.db, bc.params, bc.postgres, bc.snapshot)
+	if err != nil {
+		return nil, errors.Wrapf(err, "GetUncommittedTipView: Problem initializing UtxoView")
+	}
+	if committedBlockIndex == len(bc.bestChain)-1 {
+		return utxoView, nil
+	}
+	for ii := committedBlockIndex + 1; ii < len(bc.bestChain); ii++ {
+		// We need to get these blocks from the uncommitted blocks map
+		fullBlock, exists := bc.uncommittedBlocksMap[*bc.bestChain[ii].Hash]
+		if !exists {
+			return nil, errors.Errorf("GetUncommittedTipView: Block %v not found in block index", bc.bestChain[ii].Hash)
+		}
+		// TODO: Do we want/need to attach the event manager here? Do we need the utxo ops for anything?
```

Comment:
No UtxoOps needed. We also don't need to pass events via the eventManager afaik. The `GetUncommittedTipView` will be called many times for the same block so, we don't want to re-process events.